File: src/main/java/org/micoli/php/codeStyle/configuration/CodeStylesSynchronizationConfiguration.java
package org.micoli.php.codeStyle.configuration;

import io.swagger.v3.oas.annotations.media.Schema;
import org.micoli.php.configuration.models.DisactivableConfiguration;

public final class CodeStylesSynchronizationConfiguration implements DisactivableConfiguration {
    @Override
    public boolean isEnabled() {
        return enabled;
    }

    @Schema(description = "Enabler for panel of Code style synchronization", example = "true")
    public boolean enabled = false;

    public CodeStyle[] styles = new CodeStyle[] {};
}

File: src/main/kotlin/org/micoli/php/codeStyle/configuration/CodeStylesSynchronizationConfiguration.kt
package org.micoli.php.codeStyle.configuration

import io.swagger.v3.oas.annotations.media.Schema
import org.micoli.php.configuration.models.DisactivableConfiguration

class CodeStylesSynchronizationConfiguration : DisactivableConfiguration {
    override fun isDisabled(): Boolean {
        return !enabled
    }

    @Schema(description = "Enabler for panel of Code style synchronization", example = "true")
    var enabled: Boolean = false

    @Schema(description = "Array if code styles to synchronize")
    var styles: Array<CodeStyle> = arrayOf()
}
